hbase-mapreduce/src/test/java/org/apache/hadoop/hbase/PerformanceEvaluation.java

```diff
@@ -1999,6 +2032,47 @@ protected void testTakedown() throws IOException {
     }
   }
 
+  /*
+  Send random reads against fake regions inserted by MetaWriteTest
+   */
+  static class MetaRandomReadTest extends MetaTest {
+    private Random rd = new Random();
+    private RegionLocator regionLocator;
+
+    MetaRandomReadTest(Connection con, TestOptions options, Status status) {
+      super(con, options, status);
+      LOG.info("call getRegionLocation");
+    }
+
+    @Override
+    void onStartup() throws IOException {
+      super.onStartup();
+      this.regionLocator = connection.getRegionLocator(table.getName());
+    }
+
+    @Override
+    boolean testRow(final int i, final long startTime) throws IOException, InterruptedException {
+      if (opts.randomSleep > 0) {
+        Thread.sleep(rd.nextInt(opts.randomSleep));
+      }
+      HRegionLocation hRegionLocation = regionLocator.getRegionLocation(
+        getSplitKey(rd.nextInt(opts.perClientRunRows)), true);
+      LOG.debug("get location for region: " + hRegionLocation);
+      return true;
+    }
+
+    @Override
+    protected int getReportingPeriod() {
+      int period = opts.perClientRunRows / 10;
```

Comment:
Out of interest, we have several pattern of the implementations for this method. Some just returns opts.perClientRunRows, some divide by 10, some divide by 100. What's the difference here? Why we choose 10 here? Because the random read test uses 10? Not a big concern, just asking.

Reply:
I didn't implement the others but my understanding is it depends on the balance of getting more frequent reports and the overhead of IO. I chose 10 because random read test uses 10. And it works fine in my test. We might want to make it configurable in the future if we run into issues with too many IO.